From: David Bauer <mail@david-bauer.net>
Date: Tue, 19 Dec 2017 02:32:47 +0100
Subject: ar71xx: fix Archer C7 5GHz MAC-address

The TP-Link firmware uses (primary_mac-1) as MAC-address
for the 5GHz WiFi. This applies the same behaviour to LEDE.

Currently, the MAC-address is retrieved from eth1, which
does not exist on the Archer C7 v4. As a result from this,
every C7 v4 with LEDE carries the same MAC-Address on the 5GHz WiFi.

Signed-off-by: David Bauer <mail@david-bauer.net>

diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index be818b78638b741fff963c222bab7c395996c608..697fcb68d246aa445ce029440643ad5950e48f6d 100644
--- a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -53,6 +53,10 @@ board=$(ar71xx_board_name)
 case "$FIRMWARE" in
 "ath10k/cal-pci-0000:00:00.0.bin")
 	case $board in
+	archer-c7-v4)
+		ath10kcal_extract "art" 20480 2116
+		ath10kcal_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) -1)
+		;;
 	cf-e380ac-v1|\
 	cf-e380ac-v2|\
 	dlan-pro-1200-ac|\
@@ -98,7 +102,6 @@ case "$FIRMWARE" in
 		ath10kcal_extract "art" 20480 2116
 		ath10kcal_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) -2)
 		;;
-	archer-c7-v4|\
 	archer-c25-v1|\
 	tl-wdr6500-v2)
 		ath10kcal_extract "art" 20480 2116
